### PR TITLE
Fix clippy lints

### PR DESCRIPTION
**Pull request by yvan-sraka**

### CODE REVIEW

**Review by AureliaDolo — DISMISSED** [review]

**Comment by yvan-sraka** [comment]
bors merge

File: massa-consensus/src/ledger.rs [hunk 1]
```diff
@@ -77,6 +77,7 @@ impl LedgerChanges {
     }
 
     /// clone subset
+    #[must_use]
```
Comment (AureliaDolo) [inline massa-consensus/src/ledger.rs:80]: What does that do ?

Comment (yvan-sraka) [reply massa-consensus/src/ledger.rs:80]: It helps us to indicate that a function return a value that must be use! https://rust-lang.github.io/rfcs/1940-must-use-functions.html (meaning that if a developer forget to assign that return value, it's certainly a mistake)

**Comment by yvan-sraka** [comment]
Lints left are:

```
yvan@P3440 ~/massa (git)-[fix-clippy-lints] % cargo clippy      
warning: all variants have the same postfix: `Error`
  --> massa-network/src/error.rs:51:1
   |
51 | / pub enum HandshakeErrorType {
52 | |     HandshakeIdAlreadyExistError(String),
53 | |     HandshakeTimeoutError,
54 | |     HandshakeInterruptionError(String),
...  |
58 | |     IncompatibleVersionError,
59 | | }
   | |_^
   |
   = note: `#[warn(clippy::enum_variant_names)]` on by default
   = help: remove the postfixes and use full paths to the variants instead of glob imports
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#enum_variant_names

warning: this function has too many arguments (10/7)
   --> massa-network/src/network_worker.rs:232:5
    |
232 | /     pub fn new(
233 | |         cfg: NetworkSettings,
234 | |         private_key: PrivateKey,
235 | |         self_node_id: NodeId,
...   |
242 | |         version: Version,
243 | |     ) -> NetworkWorker {
    | |______________________^
    |
    = note: `#[warn(clippy::too_many_arguments)]` on by default
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#too_many_arguments

warning: `massa_network` (lib) generated 2 warnings
warning: all variants have the same postfix: `Error`
  --> massa-protocol-exports/src/error.rs:57:1
   |
57 | / pub enum HandshakeErrorType {
58 | |     HandshakeIdAlreadyExistError(String),
59 | |     HandshakeTimeoutError,
60 | |     HandshakeInterruptionError(String),
...  |
64 | |     IncompatibleVersionError,
65 | | }
   | |_^
   |
   = note: `#[warn(clippy::enum_variant_names)]` on by default
   = help: remove the postfixes and use full paths to the variants instead of glob imports
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#enum_variant_names

warning: `massa_protocol_exports` (lib) generated 1 warning
warning: this function has too many arguments (9/7)
   --> massa-protocol-worker/src/protocol_worker.rs:321:5
    |
321 | /     pub fn new(
322 | |         protocol_settings: &'static ProtocolSettings,
323 | |         operation_validity_periods: u64,
324 | |         max_block_gas: u64,
...   |
330 | |         controller_manager_rx: mpsc::Receiver<ProtocolManagementCommand>,
331 | |     ) -> ProtocolWorker {
    | |_______________________^
    |
    = note: `#[warn(clippy::too_many_arguments)]` on by default
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#too_many_arguments

warning: `massa_protocol_worker` (lib) generated 1 warning
warning: this function has too many arguments (9/7)
  --> massa-consensus/src/consensus_controller.rs:40:1
   |
40 | / pub async fn start_consensus_controller(
41 | |     cfg: ConsensusConfig,
42 | |     execution_command_sender: ExecutionCommandSender,
43 | |     execution_event_receiver: ExecutionEventReceiver,
...  |
56 | |     ConsensusError,
57 | | > {
   | |_^
   |
   = note: `#[warn(clippy::too_many_arguments)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#too_many_arguments

warning: this function has too many arguments (13/7)
   --> massa-consensus/src/consensus_worker.rs:170:5
    |
170 | /     pub async fn new(
171 | |         cfg: ConsensusConfig,
172 | |         protocol_command_sender: ProtocolCommandSender,
173 | |         protocol_event_receiver: ProtocolEventReceiver,
...   |
183 | |         staking_keys: Map<Address, (PublicKey, PrivateKey)>,
184 | |     ) -> Result<ConsensusWorker, ConsensusError> {
    | |________________________________________________^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#too_many_arguments

warning: `massa_consensus` (lib) generated 2 warnings
warning: this function has too many arguments (9/7)
   --> massa-node/src/main.rs:222:1
    |
222 | / async fn stop(
223 | |     bootstrap_manager: Option<BootstrapManager>,
224 | |     consensus_manager: ConsensusManager,
225 | |     consensus_event_receiver: ConsensusEventReceiver,
...   |
231 | |     api_public_handle: StopHandle,
232 | | ) {
    | |__^
    |
    = note: `#[warn(clippy::too_many_arguments)]` on by default
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#too_many_arguments

warning: `massa-node` (bin "massa-node") generated 1 warning
    Finished dev [unoptimized + debuginfo] target(s) in 0.10s
```

@damip this is about code quality and may be tackle in the current big refactoring!

File: massa-api/src/error.rs [hunk 2]
```diff
@@ -24,7 +24,7 @@ pub enum ApiError {
     /// massa_hash error: {0}
     MassaHashError(#[from] MassaHashError),
     /// consensus error: {0}
-    ConsensusError(#[from] ConsensusError),
+    ConsensusError(#[from] Box<ConsensusError>),
```
Comment (yvan-sraka) [inline massa-api/src/error.rs:27]: This, among others, fix https://github.com/massalabs/massa/issues/1265

**Comment by damip** [comment]
For sure, code quality lints can be processed in the refactoring. Can you create an issue and link it in the refactoring meta issue ?

**Review by AureliaDolo — COMMENTED** [review]
Why did you add must use on the clone subset methods ?

**Comment by yvan-sraka** [comment]
> Why did you add must use on the clone subset methods ?

Because it returns `Self`!

**Comment by AureliaDolo** [comment]
> 

I don't get it

**Comment by yvan-sraka** [comment]
> > > Why did you add must use on the clone subset methods ?
> > 
> > Because it returns `Self`!
> 
> I don't get it

It returns a value that **must be used**!

**Comment by AureliaDolo** [comment]
> > > > Why did you add must use on the clone subset methods ?
> > > 
> > > 
> > > Because it returns `Self`!
> > 
> > 
> > I don't get it
> 
> It returns a value, you **must use**:
> 
> ```rust
> function_that_perform_operation_in_place(&mut value); // NOT #[must_use]
> // VS
> let new_value = function_that_NOT_perform_operation_in_place(&value); // #[must_use]
> function_that_NOT_perform_operation_in_place(&value); // --> trigger a rust warning, because l-value not used
> ```

Yes but why do you thing these values are must use ?

**Comment by yvan-sraka** [comment]
> Yes but why do you thing these values are must use ?

The point is that the [l-value](https://en.wikipedia.org/wiki/Value_(computer_science)#lrvalue) must be passed to something (what you do with this, assign to a variable a function, or nothing is up to you), that's clearly the behavior we want a `clone`-like function to have!

**Review by AureliaDolo — APPROVED** [review]

**Comment by yvan-sraka** [comment]
bors merge

**Comment by yvan-sraka** [comment]
Bors merge

**Pull request merged**